ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/reduce_scatter_op.cpp

```diff
@@ -68,56 +71,69 @@ static ttnn::operations::binary::BinaryOpType convert_reduce_type_to_eltwise_typ
 
 namespace operations{
 namespace ccl{
-std::vector<Tensor> reduce_scatter_impl(
-    const std::vector<Tensor>& input_tensors,
+Tensor reduce_scatter_impl(
+    const Tensor& input_tensor,
     const ttnn::operations::binary::BinaryOpType binary_op_type,
     const uint32_t scatter_dim,
     const uint32_t num_links,
     const MemoryConfig& output_mem_config,
     const ttnn::ccl::Topology topology) {
     TT_FATAL(std::getenv("TT_METAL_SLOW_DISPATCH_MODE") == nullptr, "This op is only supported for Fast Dispatch");
 
-    std::vector<Tensor> output_tensors;
-    output_tensors.reserve(input_tensors.size());
-    std::vector<ReduceScatter> ops;
-    ops.reserve(input_tensors.size());
-    bool is_ring = topology ==ttnn::ccl::Topology::Ring;
-    for (uint32_t i = 0; i < input_tensors.size(); ++i) {
-        bool is_last_chip_in_clockwise_direction = is_ring ? false : i == (input_tensors.size() - 1);
-        bool is_last_chip_in_counter_clockwise_direction = is_ring ? false : i == 0;
-
-        std::optional<chip_id_t> receiver_device_id =
-            is_last_chip_in_clockwise_direction
-                ? std::nullopt
-                : std::optional<chip_id_t>(input_tensors[(i + 1) % input_tensors.size()].device()->id());
-        std::optional<chip_id_t> sender_device_id =
-            is_last_chip_in_counter_clockwise_direction
-                ? std::nullopt
-                : std::optional<chip_id_t>(input_tensors[i == 0 ? input_tensors.size() - 1 : i - 1].device()->id());
-        ops.emplace_back(ttnn::ReduceScatter{
-            binary_op_type,
-            scatter_dim,
-            num_links,
-            static_cast<uint32_t>(input_tensors.size()),
-            i,
-            receiver_device_id,
-            sender_device_id,
-            output_mem_config,
-            topology});
-        output_tensors.push_back(operation::run(ops[i], {input_tensors.at(i)}).at(0));
-    }
-    return output_tensors;
+    auto devices = input_tensor.get_workers();
+    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    operation::launch_op(
```

Comment:
You use next code to register this op
```
constexpr auto reduce_scatter =
    ttnn::register_operation<"ttnn::reduce_scatter", ttnn::operations::ccl::ExecuteReduceScatter>();
```

This will decorate an operation and will automatically wrap the call into a launch_op.
Can we remove launch_op from here?


see https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/decorators.hpp#L221

Reply:
See earlier comment and new issue tracking the problems: https://github.com/tenstorrent/tt-metal/issues/10756